packer/builder/azure/arm/artifact.go

```diff
@@ -3,34 +3,116 @@
 
 package arm
 
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"strings"
+)
+
 const (
 	BuilderId = "Azure.ResourceManagement.VMImage"
 )
 
-type artifact struct {
-	name string
+type Artifact struct {
+	StorageAccountLocation      string
+	StorageAccountName          string
+	StorageAccountContainer     string
+	StorageAccountBlob          string
+	Template                    string
+	ImageUri                    string
+	ImageTemplateUri            string
+	ImageUrlReadOnlySas         string
```

Comment:
I'm conflicted. :smile: My natural inclination is to call it a URL, but the SDK uses URI.  I'll settle on Uri to be consistent with the SDK.